packages/shopping-cart/src/action-promise-error.ts

```diff
@@ -0,0 +1,9 @@
+export class ActionPromiseError extends Error {
+	code: string;
+
+	constructor( message: string, code: string ) {
```

Comment:
The second parameter `code`: should it be called `type`, in-line with the `loadingErrorType` field name, and should its type be `ShoppingCartError | undefined`?

Reply:
That type won't work because it's sometimes going to be of the type `ResponseCartMessage.message` , which is `string` since we don't know what errors the endpoint might send back. This Error subclass represents two different types of errors: loading errors and server errors.

Reply:
...which, now that I say it, makes me think there should be separate error classes for each error type. I've added `CartActionConnectionError ` and `CartActionResponseError `.